### ISSUE TITLE
Not handling null 'result' value in statusCallback

### ISSUE THREAD

**Issue by TimNZ** [alexfernandez/loadtest#235]
https://github.com/alexfernandez/loadtest/blob/a0f4b30a4d4d44a2699b9d50625ba727e85d277f/lib/pool.js#L75

```
	finishRequest(client, result, error) {
		if (this.options.statusCallback) {
			result.requestIndex = this.requestIndex++
			result.instanceIndex = this.loadTest.instanceIndex
			this.options.statusCallback(error, result);

```
In my case 'error' is 'Connection error: connect ECONNREFUSED 127.0.0.1:3007', and result is null, 
resulting in 'TypeError: Cannot set properties of undefined (setting 'requestIndex')' on setting result.requestIndex

**Comment by alexfernandez** [comment]
Hi @TimNZ , sorry for the delay. Good catch, care to send a patch?

**Comment by lagouyn** [comment]
Similar issue here:

In HttpClient.finishRequest(), on a ECONNREFUSED, `result` is undefined, so for it's call to Pool.finishRequest() this crashes:

`result.requestIndex = this.requestIndex++`

**Comment by alexfernandez** [comment]
Hi @lagouyn , care to send a patch?

**Comment by colin-dumitru** [comment]
HI @alexfernandez . I've created a pull request to add a guard for this edge case: https://github.com/alexfernandez/loadtest/pull/242

**Comment by alexfernandez** [comment]
Thanks! Just published as 8.1.1.
